4_synthesize_on_k3s.sh

```diff
@@ -1,35 +1,41 @@
 #!/usr/bin/env bash
 
+# shellcheck disable=SC1091
 source lib/functions.sh
 source lib/k3s.sh
 
-check_variable "GITLAB_TOKEN" $GITLAB_TOKEN
-check_variable "GITLAB_ENDPOINT" $GITLAB_ENDPOINT
-check_variable "GITLAB_USERNAME" $GITLAB_USERNAME
-check_variable "TENANT_ID" $TENANT_ID
-check_variable "LIEUTENANT_URL" $LIEUTENANT_URL
-check_variable "LIEUTENANT_TOKEN" $LIEUTENANT_TOKEN
-check_variable "COMMODORE_SSH_PRIVATE_KEY" $COMMODORE_SSH_PRIVATE_KEY
+check_variable "GITLAB_TOKEN" "$GITLAB_TOKEN"
+check_variable "GITLAB_ENDPOINT" "$GITLAB_ENDPOINT"
+check_variable "GITLAB_USERNAME" "$GITLAB_USERNAME"
+check_variable "COMMODORE_SSH_PRIVATE_KEY" "$COMMODORE_SSH_PRIVATE_KEY"
+
+LIEUTENANT_URL=$(curl http://localhost:4040/api/tunnels --silent | jq -r '.["tunnels"][0]["public_url"]')
+check_variable "LIEUTENANT_URL" "$LIEUTENANT_URL"
+
+TENANT_ID=$(kubectl --context minikube --namespace lieutenant get tenant | grep t- | awk 'NR==1{print $1}')
+check_variable "TENANT_ID" "$TENANT_ID"
+
+LIEUTENANT_TOKEN=$(kubectl --context minikube --namespace lieutenant get secret "$(kubectl --context minikube --namespace lieutenant get sa api-access-synkickstart -o go-template='{{(index .secrets 0).name}}')" -o go-template='{{.data.token | base64decode}}')
+check_variable "LIEUTENANT_TOKEN" "$LIEUTENANT_TOKEN"
 
 # Launch K3s
 k3d cluster create projectsyn
```

Comment:
Same story as for minikube, k3d doesn't like Btrfs-on-LUKS. Here the workaround is documented in https://github.com/k3s-io/k3s/issues/471#issuecomment-491810975

```suggestion
if [ -d /dev/mapper ]; then
k3d_extra_args="-v /dev/mapper:/dev/mapper"
else
k3d_extra_args=""
fi
k3d cluster create projectsyn ${k3d_extra_args}
```

Reply:
Is it common to use Btrfs-on-LUKS?

Reply:
No idea, I'm also not sure if that's the only setup where missing devices in `/dev/mapper` become an issue, reading the comments it seems that some LVM setups experience similar issues.

Reply:
If you're ok with this, I'd rather add both this and the previous minikube information as part of the README, for "Troubleshooting" purposes, in case somebody experiments the same, would that be ok?